From 6585f7c879b047b9897641425339827e11a0aed0 Mon Sep 17 00:00:00 2001
From: Varun Sharma <vsharma@habana.ai>
Date: Mon, 27 Feb 2023 16:49:58 +0200
Subject: [PATCH] [SW-117739] greco: print boot strap values

This patch prints boot strap values at preboot.
System team requested to print boot strap values as sometimes
it is seen  gpio expander, boots occasionally with wrong values
after power cycle, and comes with value of 0xFF.
Added a print to debug this issue in Data center.

Change-Id: I45667dcb1a5177f0b7161c0b4d4c042941381197
Signed-off-by: Varun Sharma <vsharma@habana.ai>
---

diff --git a/arch/greco/arc/common.c b/arch/greco/arc/common.c
index bb29767..6a01abe 100644
--- a/arch/greco/arc/common.c
+++ b/arch/greco/arc/common.c
@@ -262,3 +262,13 @@
 	preboot_set_gpio_ex_see_bit(GPIO_EX1_DEV_ADDR);
 	preboot_set_gpio_ex_see_bit(GPIO_EX2_DEV_ADDR);
 }
+
+void preboot_read_straps(void)
+{
+	u32 boot_strap = regmap_readl(mmPSOC_GLOBAL_CONF_BOOT_STRAP_PINS_L);
+
+	preboot_puts_hex_raw("Boot strap pins low: ", boot_strap, true);
+	boot_strap = regmap_readl(mmPSOC_GLOBAL_CONF_BOOT_STRAP_PINS_H);
+	preboot_puts_hex_raw(" high: ", boot_strap, true);
+	preboot_puts("\n");
+}
